src/robot/libraries/BuiltIn.py

```diff
@@ -1700,6 +1732,17 @@ def _yield_repeated_keywords(self, times, name, args):
             self.log("Repeating keyword, round %d/%d." % (i+1, times))
             yield name, args
 
+    def _yield_repeated_keywords_until(self, maxtime, name, args):
+        repeat_round = 0
+        if time.time() >= maxtime:
```

Comment:
With the current implementation there is a possibility that if timeout is very small, `time.time() >= maxtime` is true even before the keyword is executed once. It would be safer (and in my opinion cleaner) to pass timeout, not maxtime, here and change the condition to `timeout <= 0`. After that maxtime can be set like `maxtime = time.time() + timeout`.